R/escape.R

```diff
@@ -4,14 +4,26 @@
 #' @param seuratObj A Seurat object.
 #' @param outputAssayName The name of the assay to store results
 #' @param doPlot If true, a FeaturePlot will be printed for each pathway
+#' @param geneSets Which gene sets should be fetched from MSigDB and calculated. Some recommendations in increasing computation time: H (hallmark, 50 gene sets), C8 (scRNASeq cell type markers, 830 gene sets), C2 (curated pathways, 6366 gene sets), GO:BP (GO biological processes, 7658). 
 #' @return The seurat object with results stored in an assay
 #' @export
-RunEscape <- function(seuratObj, outputAssayName = "escape.ssGSEA", doPlot = FALSE) {
-  GS.hallmark <- escape::getGeneSets(library = "H")
-
+RunEscape <- function(seuratObj, outputAssayName = "escape.ssGSEA", doPlot = FALSE, geneSets = c("H")) {
+  
+  ##GO:BP and other hierarchical gene sets require subcategories to be passed with them, so we should parse those individually and concatenate afterwards. I think each of these need to be individually supported, but I haven't confirmed. 
+  #check if all gene sets are non-hierarchical
+  if (all(geneSets %in% c("H", paste0("C", 1:8)))) {
+    GS <- escape::getGeneSets(library = geneSets)
+  } else if ("GO:BP" %in% geneSets) {
```

Comment:
@GWMcElfresh: are you sure this works like you intend? Would geneSets = c('H', 'GP:BP') work? Wouldnt it just grab the first term? I think if you make this a separate if, rather than elseif, it will work.

Reply:
If I'm understanding right, I think you're asking about fetching multiple library arguments at once? `getGeneSets(library = c("H", "C2")` works out as equivalent to `c(getGeneSets(library = "H") , getGeneSets(library = "C2"))`. With msigdb, it turns out that H + C1-8 spans all of the non-hierarchical gene sets, so the first if statement doesn't end up missing any cases. 

This only gets tripped up when you need to pass a hierarchical library (e.g. `getGeneSets(library = c("H", "C5"), subcategory = "BP")` which will yield `Error in msigdbr(species = spec_check, category = library, subcategory = subcategory) : 
  please specify only one category at a time` because H doesn't have a "BP" category.

for c("H", "GO:BP"), we parse for the hierarchical gene set (GO:BP), remove it and fetch that list explicitly, then fetch all of the remaining non-hierarchical gene sets and concatenate afterwards.

Reply:
> @GWMcElfresh: are you sure this works like you intend? Would geneSets = c('H', 'GP:BP') work? Wouldnt it just grab the first term? I think if you make this a separate if, rather than elseif, it will work.

Oh, I see what you mean now, I thought "GP:BP" was a typo, but I get it now. I'll support that.

Reply:
It's kind of a pain, but I think the asymmetry between hierarchical and non-hierarchical accession from msigdb means we need to explicitly support any gene sets that want to access the subcategory argument of `getGeneSets()`.